lib/mayaUsd/ufe/UsdTransform3dMayaXformStack.cpp

```diff
@@ -0,0 +1,708 @@
+    {"xformOp:rotateXYZ:rotateAxis", UsdTransform3dMayaXformStack::NdxRotateAxis},
+    {"!invert!xformOp:translate:rotatePivot", UsdTransform3dMayaXformStack::NdxRotatePivotInverse},
+    {"xformOp:translate:scalePivotTranslate", UsdTransform3dMayaXformStack::NdxScalePivotTranslate},
+    {"xformOp:translate:scalePivot", UsdTransform3dMayaXformStack::NdxScalePivot},
+    {"xformOp:shear",               UsdTransform3dMayaXformStack::NdxShear},
+    {"xformOp:scale",               UsdTransform3dMayaXformStack::NdxScale},
+    {"!invert!xformOp:translate:scalePivot", UsdTransform3dMayaXformStack::NdxScalePivotInverse}};
+
+//----------------------------------------------------------------------
+// Conversion functions from RotXYZ to all supported rotation attributes.
+//----------------------------------------------------------------------
+
+typedef VtValue (*CvtRotXYZToAttrFn)(double x, double y, double z);
+
+VtValue toXYZ(double x, double y, double z) {
+    // No rotation order conversion
+    VtValue v;
+    v = GfVec3f(x, y, z);
+    return v;
+}
+
+// Reorder argument RotXYZ rotation.
+template<MEulerRotation::RotationOrder DST_ROT_ORDER>
+VtValue to(double x, double y, double z) {
+    MEulerRotation eulerRot(TO_RAD(x), TO_RAD(y), TO_RAD(z), MEulerRotation::kXYZ);
+    eulerRot.reorderIt(DST_ROT_ORDER);
+    VtValue v;
+    v = GfVec3f(TO_DEG(eulerRot.x), TO_DEG(eulerRot.y), TO_DEG(eulerRot.z));
+    return v;
+}
+
+auto toXZY = to<MEulerRotation::kXZY>;
+auto toYXZ = to<MEulerRotation::kYXZ>;
+auto toYZX = to<MEulerRotation::kYZX>;
+auto toZXY = to<MEulerRotation::kZXY>;
+auto toZYX = to<MEulerRotation::kZYX>;
+
+VtValue toX(double x, double, double) {
+    VtValue v;
+    v = float(x);
+    return v;
+}
+
+VtValue toY(double, double y, double) {
+    VtValue v;
+    v = float(y);
+    return v;
+}
+
+VtValue toZ(double, double, double z) {
+    VtValue v;
+    v = float(z);
+    return v;
+}
+
+CvtRotXYZToAttrFn getCvtRotXYZToAttrFn(const TfToken& opName)
+{
+    // Can't get std::unordered_map<TfToken, CvtRotXYZToAttrFn> to instantiate.
+    static std::map<TfToken, CvtRotXYZToAttrFn> cvt = {
+        {TfToken("xformOp:rotateX"),   toX},
+        {TfToken("xformOp:rotateY"),   toY},
+        {TfToken("xformOp:rotateZ"),   toZ},
+        {TfToken("xformOp:rotateXYZ"), toXYZ},
+        {TfToken("xformOp:rotateXZY"), toXZY},
+        {TfToken("xformOp:rotateYXZ"), toYXZ},
+        {TfToken("xformOp:rotateYZX"), toYZX},
+        {TfToken("xformOp:rotateZXY"), toZXY},
+        {TfToken("xformOp:rotateZYX"), toZYX},
+        {TfToken("xformOp:orient"),    nullptr}}; // FIXME, unsupported.
+
+    return cvt.at(opName);
+}
+
+//----------------------------------------------------------------------
+// Conversion functions from all supported rotation attributes to RotXYZ.
+//----------------------------------------------------------------------
+
+typedef Ufe::Vector3d (*CvtRotXYZFromAttrFn)(const VtValue& value);
```

Comment:
Same as above, except in other direction: convert from attribute ordering to UFE RotXYZ.